### PR TITLE
Improve / refactor RPC documentation

### PR DESCRIPTION
**Pull request by thisisjoshford**
- refactored/organized RPC queries
- added additional examples for each query and sample replies
- tested all endpoints for accuracy
- added instructions on using an API request tool
- removed RPC document in development section to improve search queries
- updated links

fixes: #483

### CODE REVIEW

File: docs/api/rpc.md [hunk 1]
```diff
@@ -1,374 +1,592 @@
...
+Note: For HTTPie, params use either an object or array as a string.
 
+## Accounts / Contracts
+
+### View account
+ Returns view of account information.
+  - method: `query`
+  - params 
+     - `request_type`: `view_account`
+     - `finality`: `optimistic`, `near-final`, or `final`
```
Comment (bowenwang1996) [inline docs/api/rpc.md:43]: Let's avoid mentioning `near-final` in our documentation. It may create unnecessary confusion

File: docs/api/rpc.md [hunk 2]
```diff
@@ -381,715 +599,1999 @@ http post https://rpc.testnet.near.org jsonrpc=2.0 id=dontcare method=query \
...
-          "permission": "FullAccess"
-        },
-        "public_key": "ed25519:HzkMAfXsiw2iJ55moB5n2TXNYZFTv1fafWgySUWgE3VQ"
-      }
-    ]
-  }
+     "jsonrpc": "2.0",
+      "id": "dontcare",
+      "method": "block",
+      "params": [17821130]
```
Comment (bowenwang1996) [inline docs/api/rpc.md:658]: let's not use the deprecated syntax, use `{"block_id": 17821130}` instead

File: docs/api/rpc.md [hunk 3]
```diff
@@ -381,715 +599,1999 @@ http post https://rpc.testnet.near.org jsonrpc=2.0 id=dontcare method=query \
...
-  "jsonrpc": "2.0",
-  "result": {
-    "block_height": 288274,
-    "nonce": 1,
-    "permission": "FullAccess"
-  }
+     "jsonrpc": "2.0",
+      "id": "dontcare",
+      "method": "block",
+      "params": ["7nsuuitwS7xcdGnD9JgrE22cRB2vf2VS4yh1N9S71F4d"]
```
Comment (bowenwang1996) [inline docs/api/rpc.md:668]: same here

File: docs/api/rpc.md [hunk 4]
```diff
@@ -381,715 +599,1999 @@ http post https://rpc.testnet.near.org jsonrpc=2.0 id=dontcare method=query \
...
-Please note that, in the case of function call transactions, the query will not wait for **all** the receipts generated
-by this transaction to finish before returning. Rather, it will only wait for the its return value to finish (could be a promise)
-before returning. More specifically, consider the following function
-```rust
-pub fn transfer(receiver_id: String) {
-    Promise::new(receiver_id).transfer(10);
+</p>
+</details>
+
+### Genesis Records
```
Comment (bowenwang1996) [inline docs/api/rpc.md:1294]: this is already deprecated. Let's remove this from docs

**Review by bowenwang1996 — COMMENTED** [review]
let's remove all mention of `near-final`

File: docs/api/rpc.md [hunk 5]
```diff
@@ -381,715 +599,1838 @@ http post https://rpc.testnet.near.org jsonrpc=2.0 id=dontcare method=query \
...
-    "block_height": 1388549,
-    "logs": [],
-    "result": "ImZyb2wubmVhciI="
-  }
+     "jsonrpc": "2.0",
+      "id": "dontcare",
+      "method": "EXPERIMENTAL_changes_in_block",
+      "params": {
+            "block_id": "81k9ked5s34zh13EjJt26mxw5npa485SY4UNoPi6yYLo"
+      }
```
Comment (frol) [inline docs/api/rpc.md:831]: Many of these json blocks are off by one or two spaces. Let's fix them

File: docs/api/rpc.md [hunk 6]
```diff
@@ -381,715 +599,1838 @@ http post https://rpc.testnet.near.org jsonrpc=2.0 id=dontcare method=query \
...
-
-`gas_price(null | block_height: u64 | block_hash: string)`
+## State changes
+Please see our [experimental methods](/docs/api/rpc-experimenta) for more information on querying state changes.
 
-Queries for gas price using `null` (for latest) as well as a specific `block_height` or `block_hash`.
+## Transactions
+### Send transaction (async)
+### Send transaction (wait)
+### Transaction Status
```
Comment (frol) [inline docs/api/rpc.md:1304]: TODO?

File: docs/api/rpc-experimental.md [hunk 7]
```diff
@@ -7,81 +7,18 @@ sidebar_label: Experimental Methods
...
 
-Experimental API methods are feature-complete, but we do not guarantee them to be backward-compatible or exist forever. For example, we are going to remove `EXPERIMENTAL_genesis_records` API soon and `EXPERIMENTAL_genesis_config` does not have a stable structure (we used to change it quite frequently); we are committed to never do such changes for the non-EXPERIMENTAL endpoints. 
+Experimental API methods are feature-complete, but we do not guarantee them to be backward-compatible or exist forever. 
 
-EXPERIMENTAL endpoints are useful for API evaluation on the real network. While the API is in EXPERIMENTAL state we collect the feedback and can iterate on the input/output structure.
+These endpoints are useful for API evaluation on the real network. While the API is in EXPERIMENTAL state, we collect feedback and can iterate on the input/output structure.
 
 </blockquote>
 
-## Changes in Block
```
Comment (frol) [inline docs/api/rpc-experimental.md:16]: Do we deprecate rpc-experimental? I see some methods being moved to rpc.md while others are left intact.

File: docs/roles/developer/contracts/rpc.md [hunk 8]
```diff
@@ -1,42 +0,0 @@
----
-id: rpc
-title: RPC
-sidebar_label: RPC
----
-
-The generic solution to invoking smart contracts on the NEAR Platform is our RPC interface. This method is used by `near-api-js`. Our [RPC interface](/docs/api/rpc) uses Standard JSON RPC 2.0 across the board.
```
Comment (frol) [inline docs/roles/developer/contracts/rpc.md:7]: I think it makes sense to have this page under the developer role, so people can navigate through the menu while they are in the developer context. I would even argue that we need a similar page in the documentation for other roles as well for better discoverability.

P.S. the page don't need to provide the details (I think we want to move the diagram to the RPC documentation itself)

File: docs/api/rpc.md (hunk 6 above)
Comment (thisisjoshford) [reply docs/api/rpc.md:1304]: correct, currently working on this.

File: docs/api/rpc-experimental.md (hunk 7 above)
Comment (thisisjoshford) [reply docs/api/rpc-experimental.md:16]: Yes, @bowenwang1996 asked me to consolidate to one document. The ones left in `rpc-experimental.md` still need example snippets created but was told I can hold off on that for now.

Comment (frol) [reply docs/api/rpc-experimental.md:16]: I see, I just found it confusing that `EXPERIMENTAL_changes_in_block` documentation has been moved while `EXPERIMENTAL_changes` left behind :thinking:

File: docs/api/rpc.md (hunk 6 above)
Comment (thisisjoshford) [reply docs/api/rpc.md]: Completed @frol :)

File: docs/api/rpc-experimental.md (hunk 7 above)
Comment (thisisjoshford) [reply docs/api/rpc-experimental.md:16]: Totally understand, it will be my goal to get this moved over in the very near future :)

File: docs/api/rpc.md [hunk 9]
```diff
@@ -1,323 +1,132 @@
...
-    ]
-  },
-  "id": "dontcare"
-}
-```
-
-## Send transaction (async)
-
-`broadcast_tx_async(signed_transaction: string)`
+After that is setup, just copy/paste the `JSON object` example snippets below into the `body` of your request, on Postman, and click `send`.
```
Comment (bowenwang1996) [inline docs/api/rpc.md:27]:
```suggestion
After that is set up, just copy/paste the `JSON object` example snippets below into the `body` of your request, on Postman, and click `send`.
```

File: docs/api/rpc-experimental.md [hunk 10]
```diff
@@ -232,567 +169,3 @@ http post https://rpc.testnet.near.org jsonrpc=2.0 id=dontcare \
...
-            }
-        ]
-    }
-}
-```
-
-</p>
-</details>
-
-## Transaction Status with Receipts
```
Comment (bowenwang1996) [inline docs/api/rpc-experimental.md:547]: why is this removed?

Comment (thisisjoshford) [reply docs/api/rpc-experimental.md:547]: This was supposed to be moved to `rpc.md` not removed entirely. My mistake, I will fix it :)

File: docs/roles/developer/contracts/api.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+id: api 
+title: API
+sidebar_label: API
+---
+
+The generic solution to invoking smart contracts on the NEAR Platform is our RPC interface. This method is used by `near-api-js`. Our [RPC interface](/docs/api/rpc) uses Standard JSON RPC 2.0 across the board.
+
+For `view` methods:
+
+The documentation linked above includes details of the `query` method which supports the `call/<account_id>/<method name>` to call `<method name>` on the contract identified by `<account_id>`.  `<method_name>` may only be invoked as a `view` method in this case.
```
Comment (frol) [inline docs/roles/developer/contracts/api.md:11]: This should refer to the new style of `query` API (`{"request_type": "call_function"}`), and also link to the page.

File: docs/roles/developer/contracts/api.md [hunk 12]
```diff
@@ -0,0 +1,15 @@
+---
+id: api 
+title: API
+sidebar_label: API
+---
+
+The generic solution to invoking smart contracts on the NEAR Platform is our RPC interface. This method is used by `near-api-js`. Our [RPC interface](/docs/api/rpc) uses Standard JSON RPC 2.0 across the board.
+
+For `view` methods:
```
Comment (frol) [inline docs/roles/developer/contracts/api.md:9]: It is worth linking readers to the page where we describe the difference between view and change methods

File: website/sidebars.json [hunk 13]
```diff
@@ -74,7 +74,7 @@
           "roles/developer/contracts/cli",
           "roles/developer/contracts/rust",
           "development/calling-smart-contracts",
-          "roles/developer/contracts/rpc"
+          "roles/developer/contracts/api"
```
Comment (frol) [inline website/sidebars.json:77]: Reminder: we need to have proper redirects when we move pages

How is that configured? Is it completely on Render side?

**Review by frol — APPROVED** [review]
This is a much better structure! Thank you for this revamp!

File: docs/roles/developer/contracts/api.md (hunk 12 above)
Comment (thisisjoshford) [reply docs/roles/developer/contracts/api.md:9]: > This is a much better structure! Thank you for this revamp!
thank you soooo much @frol! really appreciate it :)

Will link the readers to the view / change method information as well.

File: website/sidebars.json (hunk 13 above)
Comment (thisisjoshford) [reply website/sidebars.json:77]: On it! Got a list of URL forwardings that will need to take place. And yes, Render is currently handling all of this.

File: docs/api/rpc-experimental.md [hunk 14]
```diff
@@ -7,81 +7,18 @@ sidebar_label: Experimental Methods
 <blockquote class="warning">
 <strong>heads up</strong><br><br>
 
-Experimental API methods are feature-complete, but we do not guarantee them to be backward-compatible or exist forever. For example, we are going to remove `EXPERIMENTAL_genesis_records` API soon and `EXPERIMENTAL_genesis_config` does not have a stable structure (we used to change it quite frequently); we are committed to never do such changes for the non-EXPERIMENTAL endpoints. 
+Experimental API methods are feature-complete, but we do not guarantee them to be backward-compatible or exist forever. 
 
-EXPERIMENTAL endpoints are useful for API evaluation on the real network. While the API is in EXPERIMENTAL state we collect the feedback and can iterate on the input/output structure.
+These endpoints are useful for API evaluation on the real network. While the API is in EXPERIMENTAL state, we collect feedback and can iterate on the input/output structure.
```
Comment (bowenwang1996) [inline docs/api/rpc-experimental.md:12]: I don't understand why you move most of the experimental methods out to another file but still keep the changes endpoint here. Seems rather inconsistent to me

Comment (thisisjoshford) [reply docs/api/rpc-experimental.md:12]: In slack, you told me to skip the `changes` endpoint for now. I agree that it appears incomplete this way. Should I just work on incorporating this endpoint into the new refactored `rpc.md` now?

Comment (bowenwang1996) [reply docs/api/rpc-experimental.md:12]: I think you misunderstood me. I meant that you don't have to try every combination of things to generate examples. I did not mean that you should ignore it.

Comment (thisisjoshford) [reply docs/api/rpc-experimental.md:12]: Understood, will work on this now.

File: docs/api/rpc.md [hunk 15]
```diff
@@ -1,974 +1,3433 @@
...
...hBgsgBCACNgJQDAQLIARBAzYCKCAEQa+RwYAANgIkIARBAjsBICAEQej5wIAANgJQQQEhBkEAIQgMAwsgBEEDNgIoIARBrJHBgAA2AiQgBEECOwEgIAQgAjYCUAwCC0EBIQYgBEEBNgIoIARB6PnAgAA2AiQLIAQgDDYCUAsgBEHcAGogBjYCACAEIAg2AlQgBCAEQSBqNgJYIAAgBEHQAGoQuoeAgAAhBiAEQYABaiSAgICAACAGDwsgDEERQZyRwYAAEJOHgIAAAAtBvI7BgABBIUHUkMGAABCRh4CAAAALQeSQwYAAQR9BhJHBgAAQkYeAgAAAC5UCAQJ/I4CAgIAAQRBrIgIkgICAgAAgAkEANgIMAkACQAJAAkAgAUGAAUkNACABQYAQSQ0BIAJBDGohAyABQYCABE8NAiACIAFBP3FBgAFyOgAOIAIgAUEMdkHgAXI6AAwgAiABQQZ2QT9xQYABcjoADUEDIQEMAwsgAiABOgAMIAJBDGohA0EBIQEMAgsgAiABQT9xQYABcjoADSACIAFBBnZBwAFyOgAMIAJBDGohA0ECIQEMAQsgAiABQT9xQYABcjoADyACIAFBEnZB8AFyOgAMIAIgAUEGdkE/cUGAAXI6AA4gAiABQQx2QT9xQYABcjoADUEEIQELIAAgAyABELKHgIAAIQEgAkEQaiSAgICAACABC3EBAX8jgICAgABBIGsiAiSAgICAACACIAA2AgQgAkEIakEQaiABQRBqKQIANwMAIAJBCGpBCGogAUEIaikCADcDACACIAEpAgA3AwggAkEEakGolsGAACACQQhqEKGHgIAAIQEgAkEgaiSAgICAACABCxEAIAAoAgAgASACELKHgIAAC5wCAQJ/I4CAgIAAQRBrIgIkgICAgAAgACgCACEAIAJBADYCDAJAAkACQAJAIAFBgAFJDQAgAUGAEEkNASACQQxqIQMgAUGAgARPDQIgAiABQT9xQYABcjoADiACIAFBDHZB4AFyOgAMIAIgAUEGdkE/cUGAAXI6AA1BAyEBDAMLIAIgAToADCACQQxqIQNBASEBDAILIAIgAUE/cUGAAXI6AA0gAiABQQZ2QcABcjoADCACQQxqIQNBAiEBDAELIAIgAUE/cUGAAXI6AA8gAiABQRJ2QfABcjoADCACIAFBBnZBP3FBgAFyOgAOIAIgAUE
```
Comment (bowenwang1996) [inline docs/api/rpc.md:1031]: probably doesn't make sense to include everything here. It is too large

Comment (thisisjoshford) [reply docs/api/rpc.md:1031]: Yeah, I was wondering about this one. I'll shorten it up :)

**Review by frol — APPROVED** [review]
